test/prob/test_fixture_ccdf_log.hpp

```diff
@@ -700,6 +700,249 @@ class AgradCcdfLogTestFixture : public ::testing::Test {
+                        : p4s.front());
+      p5s.push_back((is_vector<T5>::value)
+                        ? get_param<Scalar5>(parameters[n], 5)
+                        : p5s.front());
+
+      single_ccdf_log += TestClass.template ccdf_log<Scalar0, Scalar1, Scalar2,
+                                                     Scalar3, Scalar4, Scalar5>(
+          p0s.back(), p1s.back(), p2s.back(), p3s.back(), p4s.back(),
+          p5s.back());
+    }
+
+    add_var(vector_vars, p0);
+    if (is_vector<T0>::value)
+      add_var(scalar_vars, p0s);
+    else
+      add_var(scalar_vars, p0s.front());
+
+    add_var(vector_vars, p1);
+    if (is_vector<T1>::value)
+      add_var(scalar_vars, p1s);
+    else
+      add_var(scalar_vars, p1s.front());
+
+    add_var(vector_vars, p2);
+    if (is_vector<T2>::value)
+      add_var(scalar_vars, p2s);
+    else
+      add_var(scalar_vars, p2s.front());
+
+    add_var(vector_vars, p3);
+    if (is_vector<T3>::value)
+      add_var(scalar_vars, p3s);
+    else
+      add_var(scalar_vars, p3s.front());
+
+    add_var(vector_vars, p4);
+    if (is_vector<T4>::value)
+      add_var(scalar_vars, p4s);
+    else
+      add_var(scalar_vars, p4s.front());
+
+    add_var(vector_vars, p5);
+    if (is_vector<T5>::value)
+      add_var(scalar_vars, p5s);
+    else
+      add_var(scalar_vars, p5s.front());
+
+    calculate_gradients_1storder(single_gradients1, single_ccdf_log,
+                                 scalar_vars);
+    calculate_gradients_2ndorder(single_gradients2, single_ccdf_log,
+                                 scalar_vars);
+    calculate_gradients_3rdorder(single_gradients3, single_ccdf_log,
+                                 scalar_vars);
+
+    T_return_type multiple_ccdf_log
+        = TestClass.template ccdf_log<T0, T1, T2, T3, T4, T5>(p0, p1, p2, p3,
+                                                              p4, p5);
+
+    calculate_gradients_1storder(multiple_gradients1, multiple_ccdf_log,
+                                 vector_vars);
+    calculate_gradients_2ndorder(multiple_gradients2, multiple_ccdf_log,
+                                 vector_vars);
+    calculate_gradients_3rdorder(multiple_gradients3, multiple_ccdf_log,
+                                 vector_vars);
+
+    stan::math::recover_memory();
+
+    if (stan::math::is_inf(stan::math::value_of_rec(single_ccdf_log))
+        && stan::math::value_of_rec(single_ccdf_log)
+               == stan::math::value_of_rec(multiple_ccdf_log)) {
+      return;
+    }
+
+    EXPECT_NEAR(stan::math::value_of_rec(single_ccdf_log),
+                stan::math::value_of_rec(multiple_ccdf_log), 1e-7)
+        << "cdf_log evaluated in loop should match vectorized equivalent";
+
+    size_t pos_single = 0;
+    size_t pos_multiple = 0;
+    if (!is_constant_all<T0>::value && !is_empty<T0>::value
+        && !std::is_same<Scalar0, fvar<double>>::value
+        && !std::is_same<Scalar0, fvar<fvar<double>>>::value)
```

Comment:
Why not testing the gradients of fvars?

Reply:
Good point. I got rid of these and added `fvar<fvar<double>>` to the 2nd order gradient tests.

It makes me nervous that this test framework is not tested.